### PR TITLE
Enable template localization on dotnet CLI

### PR DESCRIPTION
**Pull request by rokonec**
## Background:
 Template engine (`dotnet new`) added support for template localizations in .NET 6.0. This new system replaces the old way of localizing templates that only worked on Visual Studio and works on both .NET CLI as well as VS.

This PR introduces changes to switch to the new template localization system.

## Summary of the changes
- Adds `<LocalizeTemplates>true</>` to project containing templates.
- Include PackageReference for "Microsoft.TemplateEngine.Tasks" to project containing templates.
- Generates localization files to be translated by loc team

## What to expect after merging
Every time there is a change to one of the templates:
- The dev making the change should build (at the very least) the modified template project. This will update the loc files on the local working copy,
- Push the loc files together with the template modifications. Review & merge.
- OneLocBuild integration will automatically pick up the changes and will send them for translation.
- You will receive a PR containing the translated template loc files when they are ready. Review & merge.

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -12,6 +12,10 @@
       <Uri>https://github.com/dotnet/runtime</Uri>
       <Sha>63715256cc94963d608ef3d683d2529332c5140c</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.TemplateEngine.Tasks" Version="7.0.100-alpha.1.21601.1">
```
Comment (jonathanpeppers) [inline eng/Version.Details.xml:15]: Do we need to add a Maestro subscription for this? Can it be a dependency of dotnet/installer, so when we get updates for dotnet/installer this number follows along?

See an example here on how ILLink updates along with dotnet/installer: https://github.com/xamarin/xamarin-android/pull/6670

Comment (vlada-shubina) [reply eng/Version.Details.xml:15]: Adding dependency is possible, but updates will be coming quite frequently - in templating all the packages are produced on any change (even unrelated to package).

Alternative is to add the task via Arcade SDK,  I see the repo has already reference to it https://github.com/xamarin/xamarin-android/blob/71e6ecd67b08394482317c94db9ff35dc051ffbc/global.json#L7 and dependency https://github.com/xamarin/xamarin-android/blob/71e6ecd67b08394482317c94db9ff35dc051ffbc/eng/Version.Details.xml#L21-L24   

But this version is very old, we need to update it. Also template localization task is only available from arcade SDK 7, if you need to stay on 6 - we might need to backport it.

Comment (rmarinho) [reply eng/Version.Details.xml:15]: yeah need to know for us too.

Comment (jonathanpeppers) [reply eng/Version.Details.xml:15]: @pjcollins should we setup a weekly subscription for dotnet/arcade?

We might also need to configure a few things so the version number is used throughout our build.

**Comment by vlada-shubina** [comment]
@jonathanpeppers @pjcollins another problem to solve is how to handoff/handback generated json files to OneLocBuild.
does this repo has already an pipeline to do that? I couldn't find one. 
arcade has yaml / script for this which already handles template localization, but since this repo is not on arcade, we need custom steps here.

**Comment by vlada-shubina** [comment]
> @jonathanpeppers @pjcollins another problem to solve is how to handoff/handback generated json files to OneLocBuild. does this repo has already an pipeline to do that? I couldn't find one. arcade has yaml / script for this which already handles template localization, but since this repo is not on arcade, we need custom steps here.

To finalize the effort, we need to include also similar script here https://github.com/xamarin/xamarin-macios/pull/14213/commits/477757f54970b39c4a3d01e74da5939b9192a99c.
At the build time the `templatestrings.en.json` should be renamed to `templatestrings.json` and added to localization project JSON. Could you please advise at which step it would be best to do it? Thanks

**Comment by jonathanpeppers** [comment]
@vlada-shubina could we do this in MSBuild in this project?

https://github.com/xamarin/xamarin-android/blob/main/src/Microsoft.Android.Templates/Microsoft.Android.Templates.csproj

And make a target that runs before `Build` or `Pack`?

**Comment by vlada-shubina** [comment]
> @vlada-shubina could we do this in MSBuild in this project?
> 
> https://github.com/xamarin/xamarin-android/blob/main/src/Microsoft.Android.Templates/Microsoft.Android.Templates.csproj
> 
> And make a target that runs before `Build` or `Pack`?

Yes, that's also possible. It should be done after `Build` as the translation files won't be generated before that. 
Also it should be only applied to CI build - `templatestrings.json` should not be included to repo. Is there any property that allows to distinguish CI build from others?

**Comment by jonathanpeppers** [comment]
@vlada-shubina  you should be able to check if `$(RunningOnCI)` is true in MSBuild.

**Comment by pjcollins** [comment]
@vlada-shubina @rokonec we've started running OneLocBuild as of https://github.com/xamarin/xamarin-android/commit/fa9629db3b793edc1305cce493bf946f475b6bb2. However, the repo is still pending onboarding via https://ceapex.visualstudio.com/CEINTL/_workitems/edit/663104. Do we need to wait for that process to complete before picking this work up again, or is the OneLocBuild job addition enough to unblock this?

**Comment by vlada-shubina** [comment]
@pjcollins 

it should be enough to have the job added to start it, however during onboarding you should mention that you'll be localizing templates as well, afaik this needs additional setup on their side as well.

Are you using arcade scripts for localization or do you have your own?
If you have your own scripts, you might need to add extra steps to them similar to https://github.com/xamarin/xamarin-macios/commit/af30c263816dff38c1bf0373175697539b07158e

Basically the following steps are needed:
- `templatestrings.en.json` files should be renamed to `templatestrings.json` 
- `templatestrings.json` files should be included to LocProject.json that is being handed over to translations.

On localization hand back you need to ensure that translated files are put to localize folder in .template.config folder.

If you use arcade scripts, they already do all steps above.

**Comment by pjcollins** [comment]
I've pulled these changes into https://github.com/xamarin/xamarin-android/pull/7248, I think we'll be able to close this in favor of that new one.

**Comment by jonpryor** [comment]
Superseded by PR #7248, which has been merged.

Thanks!
